File: 1557-CheckIfAStringContainsAllBinaryCodesOfSizeK/1557-CheckIfAStringContainsAllBinaryCodesOfSizeK.java
// Last updated: 3/30/2026, 2:12:28 PM
class Solution {
    public boolean hasAllCodes(String s, int k) {
        HashSet<String> map = new HashSet<>();
        for(int left = 0 ; left+k <= s.length() ; left++){
            map.add(s.substring(left,left+k));
        }
        if(map.size() == Math.pow(2,k)) return true;
        return false;
    }
}

File: 1557-CheckIfAStringContainsAllBinaryCodesOfSizeK/1557-CheckIfAStringContainsAllBinaryCodesOfSizeK.cpp
// Last updated: 3/30/2026, 2:12:23 PM
class Solution {
public:
    bool hasAllCodes(string s, int k) {
        set<string> set;
        for(int left = 0 ; left+k <= s.size() ; left++){
            set.insert(s.substr(left,k));
        }
        if(set.size()==round(pow(2,k))) return true;
        return false;
    }
};